mne/epochs.py

```diff
@@ -662,7 +662,7 @@ def __next__(self, *args, **kwargs):
     def __hash__(self):
         if not self.preload:
             raise RuntimeError('Cannot hash epochs unless preloaded')
-        return object_hash(dict(info=self.info, data=self._data))
+        return object_hash(dict(info=hash(self.info), data=self._data))
```

Comment:
I would still try to isolate the change. We should not have to modify classes for this.

To fix it I would modify the object_hash function replacing

`isinstance(x, dict)`

by

`isinstance(x, dict) or hasattr(x, 'keys')`